server/api/main.py

```diff
@@ -403,17 +403,33 @@ async def _initiate_logs_collection(start_logs_limit: asyncio.Semaphore):
     """
     db_session = await fastapi.concurrency.run_in_threadpool(create_session)
     try:
-        # we don't want initiate logs collection for aborted runs
-        run_states = mlrun.common.runtimes.constants.RunStates.all()
-        run_states.remove(mlrun.common.runtimes.constants.RunStates.aborted)
-
         # list all the runs in the system which we didn't request logs collection for yet
         runs_uids = await fastapi.concurrency.run_in_threadpool(
             get_db().list_distinct_runs_uids,
             db_session,
             requested_logs_modes=[False],
             only_uids=True,
-            states=run_states,
+            states=mlrun.common.runtimes.constants.RunStates.non_terminal_states(),
+        )
+
+        last_update_time = datetime.datetime.now(
+            datetime.timezone.utc
+        ) - datetime.timedelta(
+            seconds=int(config.runtime_resources_deletion_grace_period)
+        )
+
+        # aborted means the pods were deleted and logs were already fetched
```

Comment:
I'd add another comment here to make it clear why we have this extra query:
```suggestion
        # add all the completed/failed runs in the system which we didn't request logs collection for yet.
        # aborted means the pods were deleted and logs were already fetched.
```

And I'd add to the previous query a comment saying it's querying the still running runs (added a separate comment there)